src/Types.jl

```diff
@@ -58,17 +62,20 @@ mutable struct Config{FT<:AbstractFloat, MT<:AbstractMesh, ST<:AbstractSearch, P
     search::ST
 
     mesh::MT
-    meshscale::Vector{FT}
 
-    num_procs::Int
-    max_simultanious_evaluations::Int
+    num_threads::Int
+    max_simultaneous_evaluations::Int
+    parallel_lock::Threads.AbstractLock
```

Comment:
Can we make this a concrete type? Having abstract types in code can cause performance problems when we constantly call into functions using them (like the locking/unlocking functions).

Reply:
I have just done this and made them `ReentrantLock` concrete types.